src/main/java/net/glowstone/util/compiler/EvalTask.java

```diff
@@ -0,0 +1,185 @@
+package net.glowstone.util.compiler;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import javax.tools.FileObject;
+import javax.tools.ForwardingJavaFileManager;
+import javax.tools.JavaCompiler;
```

Comment:
Can we build this as a nested jar or something, and avoid trying to load it on JVMs that don't have the `java.compiler` module (JDK 9+) or are the compact profiles (JDK 8), rather than breaking compatibility with those JVMs?